tests/integration/test_mpfa_mpsa_partial_update.py

```diff
@@ -737,6 +737,1036 @@ def _test_one_cell_a_time_node_keyword(self):
+        )
+
+    def test_biot(self):
+        self.setup()
+
+        g, g_larger = self.g, self.g_larger
+
+        specified_data = {"inverter": "python", "biot_alpha": 1}
+
+        mechanics_keyword = "mechanics"
+        flow_keyword = "flow"
+        data_small = pp.initialize_default_data(
+            g, {}, mechanics_keyword, specified_parameters=specified_data
+        )
+
+        def add_flow_data(g, d):
+            d[pp.DISCRETIZATION_MATRICES][flow_keyword] = {}
+            d[pp.PARAMETERS][flow_keyword] = {
+                "bc": pp.BoundaryCondition(g),
+                "second_order_tensor": pp.SecondOrderTensor(np.ones(g.num_cells)),
+                "bc_values": np.zeros(g.num_faces),
+                "inverter": "python",
+                "mass_weight": np.ones(g.num_cells),
+                "biot_alpha": 1,
+            }
+
+        discr = pp.Biot(mechanics_keyword=mechanics_keyword, flow_keyword=flow_keyword)
+
+        add_flow_data(g, data_small)
+
+        discr.discretize(g, data_small)
+        # Discretization on a small problem
+
+        # Perturb one node
+        g_larger.nodes[0, 2] += 0.2
+        # Faces that have their geometry changed
+        update_faces = np.array([2, 21, 22])
+
+        # Perturb the permeability in some cells on the larger grid
+        mu, lmbda = np.ones(g_larger.num_cells), np.ones(g_larger.num_cells)
+
+        high_coeff_cells = np.array([7, 12])
+        stiff_larger = pp.FourthOrderTensor(mu, lmbda)
+
+        specified_data_larger = {"fourth_order_tensor": stiff_larger, "biot_alpha": 1}
+
+        # Do a full discretization on the larger grid
+        data_full = pp.initialize_default_data(
+            g_larger, {}, mechanics_keyword, specified_parameters=specified_data_larger
+        )
+        add_flow_data(g_larger, data_full)
+
+        discr.discretize(g_larger, data_full)
+
+        # Cells that will be marked as updated, either due to changed parameters or
+        # the newly defined topology
+        update_cells = np.union1d(self.new_cells, high_coeff_cells)
+
+        updates = {
+            "modified_cells": update_cells,
+            #            "modified_faces": update_faces,
+            "map_cells": self.cell_map,
+            "map_faces": self.face_map,
+        }
+
+        # Data dictionary for the two-step discretization
+        data_partial = pp.initialize_default_data(
+            g_larger, {}, mechanics_keyword, specified_parameters=specified_data_larger
+        )
+        add_flow_data(g_larger, data_partial)
+        data_partial["update_discretization"] = updates
+
+        self._update_and_compare(
+            data_small,
+            data_partial,
+            data_full,
+            g_larger,
+            keywords=[flow_keyword, mechanics_keyword],
+            discr=discr,
+        )
+
+
+if __name__ == "__main__":
+    unittest.main()
+""" Unit tests for the partial update features of mpfa and mpsa.
```

Comment:
This looks really strange. Have you accidentally pasted something here?

Reply:
Yes, you're probably right. Will fix.